### ISSUE TITLE
LightmapGI quality/ray count UX improvements.

### ISSUE THREAD

**Issue by lander-vr** [godotengine/godot-proposals#9789]
### Describe the project you are working on

Lightmapper improvements

### Describe the problem or limitation you are having in your project

Changing the quality of lightmap bakes is cumbersome when the default settings do not suffice. The enum approach of having individual predefined quality settings makes it very easy to understand for new users, but very inefficient and awkward for those who want finer control.
In order to have finer control and make adjustments to the baking quality for scenes where the presets do not suffice, the user must open project settings to change one of the ray counts settings, close it, rebake, reopen if necessary, rinse and repeat...

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I've got two possible (similar) proposals for this:

#### 1. Quality scalar:
Replace the quality enum property in the lightmapGI node with a quality float that scales a predefined amount of ray count (256 seems like a decent baseline):
Easy UX around changing bake quality, no need to repeatedly open project settings to make adjustments. It scaling 256 keeps it very accessible and easy to understand: higher values mean better quality/less noise, lower values mean worse quality/more noise.

#### 2. "Sample count" approach:
An alternative approach could be using an INT that calculates ray count based on 2^x. This makes it so ray count is always a power of 2. This could make sense because early samples have a higher impact than later samples, meaning you need larger differences at higher sample counts for equal quality differences. i.e. 16 - > 128 will have a larger impact on convergence than 2048 -> 4096, despite the ray count difference being smaller. This would mean you'd need to set a value of 5 to get a ray count of 32, 6 for 64, 7 for 128, etc...
This could be slightly weird though since the minimum ray count is 16, which would effectively limit this property to a minimum value of 4. The bake times would also increase ~exponentially with each added integer.

I think either approaches could work, for approach 1 it may could be worth adding a tool-tip stating it's preferred to keep increases in values as a power of 2, but I'm not sure how important that actually is.

I think this proposal keeps the lightmapper quality setting easy to understand while giving much more (accessible) control.

#### 1. Quality scalar:
The quality setting turns into a float property converting into a ray count by `ray_count = 256 * quality`

#### 2. "Sample count" approach:
The quality setting turns into an int property converting into a ray count by `ray_count = 2^quality`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It cannot be worked around.

### Is there a reason why this should be core and not an add-on in the asset library?

Workflow improvements to core lightmapper.

**Comment by Calinou** [comment]
I agree with the quality scalar approach, but the issue is implementing this in a way that doesn't break compatibility with existing projects. If there's a significant mismatch in the ray count, existing lightmaps will continue to look correct, but they will look suddenly different as soon as you bake them after upgrading a project.

We can read the existing quality value and interpret it as a quality scalar (e.g. Ultra becomes `ultra_ray_count / 128` where 128 is the default Medium ray count). This will probably result in the expected ray count in most projects.

![image](https://github.com/godotengine/godot-proposals/assets/180032/11ba7b55-a025-4188-acd3-ac8de5ddb978)

**Comment by lander-vr** [comment]
@Calinou I'd suggest to have a separate scalar for the probe quality. I think probes can probably often get away with lower ray counts, so I wouldn't sacrifice the individual control of these parameters just to have them grouped into one property, they drive different settings after all. I think one parameter being called "quality" or "bake quality" and the other "probe quality" would be self explanatory enough (especially with docs and a tooltip). It would fit in the gen probes group, though I'd argue that group is overkill/unnecessary and the two settings that would be under gen probes fit better in the tweaks group, since they are also settings you "tweak" for baking.
